pkg/pillar/types/assignableadapters.go

```diff
@@ -516,14 +609,47 @@ func (aa *AssignableAdapters) CheckParentAssigngrp() bool {
 	for i := range aa.IoBundleList {
 		ioBundle := &aa.IoBundleList[i]
 		if ioBundle.AssignmentGroup == cycleDetectedAssigngrp {
-			ioBundle.Error = "Cycle detected, please check provided parentassigngrp/assigngrp"
-			ioBundle.ErrorTime = time.Now()
+			ioBundle.Error.Append(cycleDetectedError{})
 		}
 	}
 
 	return true
 }
 
+type ioBundleCollision struct {
+	phylabel   string
+	usbaddr    string
+	usbproduct string
+	pcilong    string
+	assigngrp  string
+}
+
+func (i ioBundleCollision) String() string {
+	return fmt.Sprintf("phylabel %s - usbaddr: %s usbproduct: %s pcilong: %s assigngrp: %s", i.phylabel, i.usbaddr, i.usbproduct, i.pcilong, i.assigngrp)
+}
+
+type ioBundleCollisionErr struct {
+	collisions []ioBundleCollision
+}
+
+func (i ioBundleCollisionErr) Error() string {
+	collisionErrStrPrefix := "ioBundle collision:"
+
+	collisionStrs := make([]string, 0, len(i.collisions))
+	for _, collision := range i.collisions {
+		collisionStrs = append(collisionStrs, collision.String())
+	}
+	collisionErrStrBody := strings.Join(collisionStrs, "||")
+
+	return fmt.Sprintf("%s||%s||", collisionErrStrPrefix, collisionErrStrBody)
+}
+
+func newIoBundleCollisionErr() ioBundleCollisionErr {
+	return ioBundleCollisionErr{
+		collisions: []ioBundleCollision{},
+	}
+}
+
 // CheckBadUSBBundles sets ib.Error/ErrorTime if bundle collides in regards of USB
```

Comment:
Can you update the comment to say that this re-evaluates/updates the set of errors (is it only collision errors) across all of the bundles?
THe current comment doesn't indicate that it will clear existing errors.

Reply:
now: `// CheckBadUSBBundles sets and clears ib.Error/ErrorTime if bundle collides in regards of USB`